sql-presto: Add New Recipe

kat-co opened the pull request:
### Brief summary of what the package does

Allows users to utilize [Presto](https://prestodb.io/) via SQLi.

### Direct link to the package repository

https://github.com/kat-co/sql-prestodb

### Your association with the package

It me: the maintainer.

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

Please confirm with `x`:

- [X] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [X] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [X] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [X] My elisp byte-compiles cleanly
- [X] `M-x checkdoc` is happy with my docstrings
- [X] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

purcell commented:
Looks good, and this is cool to see given that I was excited to see Presto being used to add SQL support to Apache Pulsar recently.

I note that you're generating the elisp file using a makefile, but the two are out of sync: the current generated artifact doesn't contain the `;; Package-Requires: ((emacs "24.4"))` line. Is that necessary?

In general your generation would be simpler if you had a more conventional layout in `src/sql-prestodb.el` which included the header boilerplate, with just a `$$COMMENTARY$$` placeholder or similar which the Makefile could replace. Spreading the headers across a separate file is a bit more complicated with no obvious benefit, and it makes it less convenient to work with the source file (e.g. you'd potentially get different behaviour from code evaluated there due to `lexical-binding` not being defined in it).

DamienCassou commented:
I'm closing this for now. Please reopen whenever you have made the requested changes.

kat-co commented:
Hey all, thanks for the great review. I apologize for the long delay on making the requested changes. They are pushed up under the 1.0.2 tag:

- I consider generating the comments within the `Makefile` a failed experiment and have moved them into the source code as suggested.
- I've added a `sql-presto` `defgroup` as suggested.
- I've added the `;;;###autoload` cookie as suggested.

Thanks, and I hope this clears things for landing :)

purcell commented:
That's mostly great - you just need to remove the backslashes in [this line](https://github.com/kat-co/sql-prestodb/blob/f3962ae8d437b19e7a2bb561ed7ee6f9ff27f5e0/artifacts/sql-presto.el#L7). In the meantime I'll re-open this, but please add a quick comment here when fixed so we know we can merge.

kat-co commented:
Thanks! This fix has been pushed and tagged as 1.0.3.

purcell commented:
> Thanks! This fix has been pushed and tagged as 1.0.3.

Ah, but it's not yet in `artifacts/sql-presto.el`, so I think perhaps a `make` is needed.

kat-co commented:
Dangit! Sorry, that's pushed up as 1.0.4 now :|

purcell commented:
Great - merged now! Welcome to MELPA. :tada:

kat-co commented:
:tada: 

And thanks for the warm welcome, and great reviews!
